src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AttendanceService.java

```diff
@@ -77,35 +56,64 @@ public void updateAttendance(long id, UpdateAttendanceDto attendanceDto) throws
                         .createDateTime(attendanceData.getCreateDateTime())
                         .caseNoteId(attendanceData.getCaseNoteId())
                         .build())
-                  .collect(Collectors.toSet());
+                .collect(Collectors.toSet());
     }
 
     public AbsentReasonsDto getAbsenceReasons() {
         return new AbsentReasonsDto(AbsentReason.getPaidReasons(), AbsentReason.getUnpaidReasons());
     }
 
+    @Transactional
+    public void createAttendance(final CreateAttendanceDto attendanceDto) {
 
-    private void applyAttendanceWorkflow(String offenderNo, Attendance attendance) {
-         final var eventOutcome = nomisEventOutcomeMapper.getEventOutcome(
-                 attendance.getAbsentReason(),
-                 attendance.isAttended(),
-                 attendance.isPaid());
+        var attendance = attendanceRepository.save(toAttendance(attendanceDto));
+        entityManager.flush();
```

Comment:
what's going on here then?

Reply:
The idea is that we do the main save, hopefully failing fast so we don't proceed with the API calls. If everything is ok, crack on with the API call's then do a final update to add the case note id. 

I'm not really convinced it's giving us much though because if it was going to fail it would for both updates unless the nature of errors is validation which should be resolved before getting to the service.

Reply:
this isn't going to do anything though.  You are inside a transaction.  The only way to commit at this point is to end the transaction and start a new one.